refactor: Improve copyFileSystemToPath function

kbudde opened the pull request:
another improvement idea for #40

fritzduchardt commented on internal/myks/util.go:
```diff
@@ -119,36 +112,21 @@ func copyFileSystemToPath(source fs.FS, sourcePath string, destinationPath strin
-		} else {
-
-			// Open the source file
-			srcFile, err := source.Open(path)
```
Are you sure this will work without source.Open reading from the embedded file system?

kbudde replied:
Good point, probably not ;)

kbudde commented:
@fritzduchardt  I'm bad in explaining in code reviews. So here another proposal for #40 following your code more closely. 

What did I change:
- Moved the named return value to WalkDirFunc; it's more usefull in the inner block.
- return early for directory -> reduces indentation as it removes the else
- simplified saveClose with errors.Join
- return fs.WalkDir instead of err := fs.Wallkdir..... return err
- combined `if err!=nil { return err} \n return nil` into `return err` at the end
